### ISSUE TITLE
MissingPluginException

### ISSUE THREAD

**Issue by kidandcat** [mannprerak2/nearby_connections#22]
I'm getting this error when running the example:

MissingPluginException(No implementation found for method startAdvertising on channel nearby_connections)

Phone model: CLT-L29
EMUI: 10.0.0
Android: 10

**Comment by mannprerak2** [comment]
Hi @kidandcat 
I am unable to reproduce this issue.
Perhaps you can try calling `flutter clean` first and then retry.
Also, ensure that you are running the latest stable version of flutter (Run `flutter upgrade`)

If that doesn’t solve your issue, please send the output of `flutter doctor` and the list of packages you are using.

**Comment by kidandcat** [comment]
# This is my nearby related code:

```
class NearbyAPI {
  static final NearbyAPI _singleton = new NearbyAPI._internal();
  final Strategy strategy = Strategy.P2P_STAR;
  Controller _c;

  factory NearbyAPI(Controller c) {
    _singleton._c = c;
    return _singleton;
  }
  NearbyAPI._internal();

  initialize() async {
    bool a = await Nearby().checkLocationPermission();
    if (!a) {
      await Nearby().askLocationPermission();
    }
  }

  advertise() async {
    try {
      await Nearby().startAdvertising(
        _c.profession.value,
        strategy,
        onConnectionInitiated: onConnectionInit,
        onConnectionResult: (String id, Status status) {
          print("connection $id: $status");
        },
        onDisconnected: (String id) {
          _c.selectedPeer = null;
        },
        serviceId: "be.galax.x",
      );
      _c.setAdvertising(true);
    } catch (e) {
      print("ERROR advertise() " + e.toString());
      _c.setAdvertising(false);
    }
  }

  discover() async {
    try {
      await Nearby().startDiscovery(
        _c.profession.value,
        strategy,
        onEndpointFound: (String id, String userName, String serviceId) {
          var p = new Peer();
          p.peerID = id;
          p.profession = userName;
          _c.nearbyDevices.add(p);
        },
        onEndpointLost: (String id) {
          _c.nearbyDevices.removeWhere((Peer p) => p.peerID == id);
        },
        serviceId: "be.galax.x",
      );
    } catch (e) {
      print(e);
    }
  }

  onConnectionInit(String id, ConnectionInfo info) {
    Nearby().acceptConnection(id, onPayLoadRecieved: (endpointId, payload) {
      if (payload.type == PayloadType.BYTES) {
        String str = String.fromCharCodes(payload.bytes);
        _c.receivedMessage(str);
      } else {
        print("Error, payload encoding not supported");
      }
    });
    // set selected peer as active
    _c.selectedPeer.value =
        _c.nearbyDevices.indexWhere((Peer p) => p.connID == id);
  }

  requestConnection(String id) {
    try {
      Nearby().requestConnection(
        _c.profession.value,
        id,
        onConnectionInitiated: onConnectionInit,
        onConnectionResult: (id, status) {},
        onDisconnected: (id) {},
      );
    } catch (exception) {
      // called if request was invalid
    }
  }

  sendMsg(Peer activePeer, String msg) {
    Nearby()
        .sendBytesPayload(activePeer.connID, Uint8List.fromList(msg.codeUnits));
  }
}
```


# Log:

```
I/flutter (11854): ERROR advertise() 'package:nearby_connections/src/nearby.dart': Failed assertion: line 209 pos 12: 'userNickName != null && strategy != null && serviceId != null': is not true.
D/AwareBitmapCacher(11854): handleInit switch not opened pid=11854
W/Settings(11854): Setting device_provisioned has moved from android.provider.Settings.Secure to android.provider.Settings.Global.
V/HiTouch_HiTouchSensor(11854): User setup is finished.
I/HwViewRootImpl(11854): removeInvalidNode all the node in jank list is out of time
W/IInputConnectionWrapper(11854): getTextBeforeCursor on inactive InputConnection
W/IInputConnectionWrapper(11854): getSelectedText on inactive InputConnection
W/IInputConnectionWrapper(11854): getTextAfterCursor on inactive InputConnection
W/IInputConnectionWrapper(11854): beginBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(11854): endBatchEdit on inactive InputConnection
I/flutter (11854): ERROR advertise() MissingPluginException(No implementation found for method startAdvertising on channel nearby_connections)
```

It is strange, because I first do a call when initializing, and it fails right (by assertions), but later I call it again and it throws that MissingPluginException.

**Comment by mannprerak2** [comment]
Okay, this is an issue with the plugin.

I'll try to fix this by today, meanwhile you can use version `1.1.1+1` that should work without any issues.

**Comment by kidandcat** [comment]
Confirm that version `1.1.1+1` works without issues.

**Comment by mannprerak2** [comment]
@kidandcat 
Seems like this was an easy fix, I have pushed a new version `2.0.2`.
Please try using that, you may want to run `flutter clean` after updating the version.

If you still have any issues regarding this, please reopen this issue.
